Symfony/CS/AbstractOrderedUseFixer.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS;
+
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * @author Sebastiaan Stok <s.stok@rollerscapes.net>
+ * @author Graham Campbell <graham@mineuk.com>
+ */
+abstract class AbstractOrderedUseFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        $namespacesImports = $tokens->getImportUseIndexes(true);
+        $usesOrder = array();
+
+        if (!count($namespacesImports)) {
+            return $content;
+        }
+
+        foreach ($namespacesImports as $uses) {
+            $uses = array_reverse($uses);
+            $usesOrder = array_replace($usesOrder, self::getNewOrder($uses, $tokens));
+        }
+
+        // First clean the old content
+        // This must be done first as the indexes can be scattered
+        foreach ($usesOrder as $use) {
+            for ($i = $use[1]; $i <= $use[2]; ++$i) {
+                $tokens[$i]->clear();
+            }
```

Comment:
use ``$tokens->clearRange($use[1], $use[2]);`` instead of reimplementing the same